lib/src/presentation/src/presentation_widget.dart

```diff
@@ -202,38 +206,56 @@ class _PresentationWidgetState extends State<PresentationWidget> {
-                // the bottom most in the stack
-                if (index == 0) {
-                  return SizedBox.expand(child: widget.background);
-                }
-                // if slide position is before to the current slide,
-                // we add the widges in stack on top of another with
-                // the current slide being at the top, them being transparent.
-                return Stack(
-                  children: [
-                    Opacity(
-                      opacity: _currentSlide.value == index - 1 ? 1 : 0,
-                      child: widget.slides[index - 1],
-                    ),
-                  ],
-                );
-              },
-            ),
+                  widget.slides.length + 2, // +2 for background and end slide
+                  (index) {
+                    // if the slide position is later to the current slide,
+                    // we return a blank widget to display nothing.
+                    if (index - 1 > _currentSlide.value) {
+                      return const SizedBox.shrink();
+                    }
+                    // show the end slide after all the slides (topmost)
+                    if (index - 1 == widget.slides.length) {
+                      // end of the presentation
+                      return const EndSlide(
+                        key: ValueKey('EndSlide'),
+                      );
+                    }
+                    // show the presentation background at
+                    // the bottom most in the stack
+                    if (index == 0) {
+                      return SizedBox.expand(child: widget.background);
+                    }
+                    // if slide position is before to the current slide,
+                    // we add the widges in stack on top of another with
+                    // the current slide being at the top, them being transparent.
+                    return Stack(
+                      children: [
+                        Opacity(
+                          opacity: _currentSlide.value == index - 1 ? 1 : 0,
+                          child: widget.slides[index - 1],
+                        ),
+                      ],
+                    );
+                  },
+                ) +
+                // show the current slide number if the flag is set to true
+                // and the current slide is not the end slide
+                // Adding it to the stack so that it is always on top
+                [
+                  widget.ShowCurrentSlideNumber == true&&_currentSlide.value < widget.slides.length
+                      ? Container(
+                        margin: EdgeInsets.only(bottom: 20),
+                          alignment: Alignment.bottomCenter,
+                          child: Text(
+                            '${_currentSlide.value + 1}/${widget.slides.length}',
+                            style: const TextStyle(
```

Comment:
Also, add the same of what textTheme we are using to show the number in the doc comments. This helps user understand and style it accordingly.

Reply:
Doc comments are documentation comments. The comments that starts with /// in case of dart.

Here, I am referring to the ones that are above the PresentationWidget class.